packages/babel-traverse/src/path/family.ts

```diff
@@ -312,6 +312,53 @@ export function getAllPrevSiblings(this: NodePath): NodePath[] {
   return siblings;
 }
 
+type digitalWithoutZero = "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9";
+type digital = "0" | digitalWithoutZero;
+
+// only support 0 to 999 (since we won't use too large number in path.get())
+type Num =
+  | `${digital}`
+  | `${digitalWithoutZero}${digital}`
+  | `${digitalWithoutZero}${digital}${digital}`;
+
+// convert "1" to 0 (string index to number index)
+type ToNumber<T extends string> = T extends Num ? 0 : T;
+
+type Pattern<Obj extends string, Prop extends string> = `${Obj}.${Prop}`;
+
+// split "body.body.1" to ["body", "body", 1]
+type Split<P extends string> = P extends Pattern<infer O, infer U>
+  ? [ToNumber<O>, ...Split<U>]
+  : [ToNumber<P>];
+
+// get all K with Node[K] is t.Node
+type NodeKeyOf<Node extends t.Node | t.Node[]> = keyof Pick<
+  Node,
+  {
+    [Key in keyof Node]-?: Node[Key] extends t.Node | t.Node[] ? Key : never;
```

Comment:
Q: What does the `-` mean?

Reply:
Means we remove the optional modifier `?` to the result. See https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers